src/Lucene.Net/Search/FieldComparator.cs

```diff
@@ -778,24 +779,24 @@ public sealed class StringOrdValComparator:FieldComparator
 		{
 			
 			private int[] ords;
-			private System.String[] values;
+			private UnmanagedStringArray values;
 			private int[] readerGen;
 			
 			private int currentReaderGen = - 1;
-			private System.String[] lookup;
+			private UnmanagedStringArray lookup;
```

Comment:
Should probably be: `UnmanagedStringArray.UnmanagedString[]` instead

Reply:
The problem is that the `lookup` is initially allocated more then needed (maxDoc + 1). 
So to pass it like `lookup = currentReaderValues.lookup` will require me to do an allocation of the actual used `UnamangedStrings`

Reply:
Why? You will copy the _reference_ to the array, not the reference itself.